src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -19,19 +20,19 @@ public class ExceptionHandlerMiddleware
         private readonly ExceptionHandlerOptions _options;
         private readonly ILogger _logger;
         private readonly Func<object, Task> _clearCacheHeadersDelegate;
-        private readonly DiagnosticSource _diagnosticSource;
+        private readonly DiagnosticListener _diagnosticListener;
```

Comment:
Change from `DiagnosticSource` to `DiagnosticListener` as second has fast enabled check (and used elsewhere for it)

How breaking is a change to a standard DI type?

Reply:
Officially, changing the constructor is breaking and we wouldn't do it in a minor or patch. However, since this is a major and the constructor is only expected to be called via DI it's likely fine.

Reply:
Does the exception handling path need to be faster?

Reply:
A lot of this is about making the success case faster.